scipy/stats/stats.py

```diff
@@ -138,80 +138,51 @@ def _chk2_asarray(a, b, axis):
     return a, b, outaxis
 
 
-def _shape_with_dropped_axis(a, axis):
+def _broadcast_array_shapes(arrays, axis=None):
```

Comment:
Could consider replacing these functions in a separate PR along with the fix in `_broadcast_concatenate`